### PR TITLE
Filter .s sources with build tags and suffixes

### LINKED ISSUE #327: Build tag filtering for `.s` files
#310 added build tag support, but the filtering step doesn't seem to be running for `.s` files, which are all still built unconditionally.

Test case here: https://github.com/nelhage-stripe/bug_rules_go/tree/master/build_tags_asm

### PR DESCRIPTION
**Pull request by jayconrod**
Fixes #327

### CODE REVIEW

File: tests/tags_os/tags_os_test.go [hunk 1]
```diff
@@ -24,3 +24,21 @@ func TestFoo(t *testing.T) {
...
+	var want int
+	switch runtime.GOOS {
+	case "darwin":
+		want = 12
+	case "linux":
+		want = 34
+	default:
+		want = 56
+	}
+	got := baz()
```
Comment (pmbethe09) [inline tests/tags_os/tags_os_test.go]: push into if

File: go/tools/filter_tags/filter_tags.go [hunk 2]
```diff
@@ -50,6 +51,7 @@ func filterFilenames(bctx build.Context, inputs []string) ([]string, error) {
 
 func main() {
 	cgo := flag.Bool("cgo", false, "Sets whether cgo-using files are allowed to pass the filter.")
+	quiet := flag.Bool("quiet", false, "Don't print filenames. Return code will be 0 if any files pass the filter.")
```
Comment (pmbethe09) [inline go/tools/filter_tags/filter_tags.go]: just realized that all of these are local to main, which I guess is fine, but I am used to flags being in the main file's 'var' section, for clarity of what flags a binary supports.

**Comment by jayconrod** [comment]
Jenkins test this please

**Comment by jayconrod** [comment]
Jenkins failure was due to #349, not caused by this PR.

**Pull request merged**